jetpack-start/wp-cli-keys.php

```diff
@@ -0,0 +1,196 @@
+			if ( $result ) {
+				WP_CLI::line( '- Added VaultPress SSH key to ' . $result );
+			} else {
+				WP_CLI::warning( '- Failed to find valid, accessible path to add SSH key' );
+			}
+		}
+
+		// TODO: set up WP=>VaultPress SSH access
+
+		WP_CLI::success( 'VaultPress configured successfully.' );
+	}
+
+	private function register_vaultpress( $api_key ) {
+		return VaultPress::register( $api_key );
+	}
+
+	private function install_vaultpress_ssh_keys( $ssh_key ) {
+		// write VP ssh key to ~/.ssh/authorized_keys
+		foreach ( array( 'authorized_keys', 'authorized_keys2' ) as $authorized_keys_file ) {
+			$authorized_keys_path = sprintf( '%s/.ssh/%s', getenv( 'HOME' ), $authorized_keys_file );
+
+			// Add the key to authorized keys if the file exists and doesn't already contain this key
+			if ( file_exists( $authorized_keys_path ) && ! strpos( file_get_contents( $authorized_keys_path ), $ssh_key ) ) {
+				file_put_contents( $authorized_keys_path, $ssh_key, FILE_APPEND | LOCK_EX );
+				return $authorized_keys_path;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Install Jetpack keys
+	 *
+	 * ## OPTIONS
+	 *
+	 * --jetpack_id=<jetpack_id>
+	 * : The Jetpack ID key provided by the Jetpack Start API.
+	 *
+	 * --jetpack_secret=<jetpack_secret>
+	 * : The Jetpack secret key provided by the Jetpack Start API.
+	 *
+	 * --jetpack_access_token=<jetpack_access_token>
+	 * : The Jetpack Access Token provided by the Jetpack Start API.
+	 *
+	 * [--jetpack_is_private]
+	 * : Add this flag if the site is not public (i.e. not accessible via the internet)
+	 *
+	 * [--jetpack_init_modules]
+	 * : Whether to initialize the default recommended modules.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp jetpack-start keys jetpack --jetpack_id=12345 --jetpack_secret="qwerty..." --jetpack_access_token="zxcv..."
+	 *
+	 * @subcommand jetpack
+	 */
+	public function install_jetpack_keys( $args, $assoc_args ) {
+		if ( ! class_exists( 'Jetpack' ) ) {
+			WP_CLI::error( 'Failed to install Jetpack keys: please activate the Jetpack plugin and re-run this script.' );
+		}
+
+		if ( ! is_user_logged_in() ) {
+			WP_CLI::error( 'Failed to install Jetpack keys: please re-run this script with the `--user` param set to the Jetpack Connection Owner' );
+		}
+
+		WP_CLI::line( 'Configuring Jetpack' );
+
+		$user_id = get_current_user_id();
+
+		$this->register_jetpack( $user_id, $assoc_args );
+
+		if ( isset( $assoc_args['jetpack_init_modules'] ) ) {
+			WP_CLI::line( '- Configuring recommended Jetpack modules' );
+			$this->init_recommended_jetpack_modules();
+		}
+
+		WP_CLI::success( 'Jetpack configured successfully.' );
+	}
+
+	private function register_jetpack( $user_id, $args ) {
+		$jetpack_id = (int) $args['jetpack_id'];
+		$jetpack_secret = (string) $args['jetpack_secret'];
+		$jetpack_access_token = (string) $args['jetpack_access_token'];
+		$jetpack_is_private = (bool) WP_CLI\Utils\get_flag_value( $args, 'jetpack_is_private', false );
+
+		Jetpack_Options::update_options(
```

Comment:
Good point. It would be nicer if Jetpack had a `register` method (similar to `VaultPress::register`) that would accept these args instead of us manually updating the options.